File: restarter.py
import subprocess
import discord
import util

def add_commands(tree):
	@tree.command(name="restartstream", description="Run this command to restart the stream if butt wont connect.", guild=discord.Object(id=util.GUILD_ID))
	async def restartstream(context):
		if(not await util.check_admin(context)):
			return
		subprocess_result = subprocess.run(["./restart_stream.sh"], capture_output=True)
		result = subprocess_result.stdout.decode("utf-8").strip()
		stderr = subprocess_result.stderr.decode("utf-8").strip()
		print("restarting stream")
		print(f"stdout:\n{result}")
		print(f"stderr:\n{stderr}")
		message = "The stream has been restarted. BUTT should reconnect." if result.endswith("ok") else "Something went wrong, ask your local web person for help."
		await context.response.send_message(message)

File: src/commands/restarter.ts
import { ApplicationCommandTypes } from "@discordeno/bot";
import { createCommand } from "../commands.ts";
import { env } from "../config.ts";

createCommand({
	name: "restarticecast",
	description: "restart icecast (if butt wont connect)",
	type: ApplicationCommandTypes.ChatInput,
	async execute(interaction) {
		const cmd = new Deno.Command(env.RESTART_STREAM_SCRIPT, { args: [] });
		const { stdout } = await cmd.output();
		const output = new TextDecoder().decode(stdout);
		const content = output.trim().endsWith("ok")
			? "Icecast has been restarted. BUTT should reconnect."
			: "Something went wrong, ask your local web person for help.";
		await interaction.respond({ content });
	},
});
